File: appln_teacherbot_assam/mod_teacherbot_assam/web_form/session_tracking_tool/session_tracking_tool.js
frappe.ready(function() {
	// bind events here
	frappe.web_form.on('district_name', (field, value) => {
		frappe.call({
			method: 'appln_teacherbot_assam.mod_teacherbot_assam.web_form.session_tracking_tool.session_tracking_tool.get_block',
			args: {
				'district_name':value
			},
			callback: function(res) {
				if (!res.exc) {
					// code snippet
					console.log(res.message)
					frappe.web_form.fields_dict['block_name'].df.options = res.message
					frappe.web_form.get_field('block_name').refresh()
				}
			}
		});
	});

	frappe.web_form.on('block_name', (field, value) => {
		district_name = frappe.web_form.get_value(["district_name"]);
		frappe.call({
			method: 'appln_teacherbot_assam.mod_teacherbot_assam.web_form.session_tracking_tool.session_tracking_tool.get_school',
			args: {
				'district_name':district_name,
				'block_name':value

			},
			callback: function(res) {
				if (!res.exc) {
					// code snippet
					console.log(res.message)
					frappe.web_form.fields_dict['school_name_and_code'].df.options = res.message
					frappe.web_form.get_field('school_name_and_code').refresh()
				}
			}
		});
	});

	frappe.web_form.on('session_planned_in_month', (field, value) => {
		frappe.call({
			method: 'appln_teacherbot_assam.mod_teacherbot_assam.web_form.session_tracking_tool.session_tracking_tool.validate_planned_session',
			args: {
				'session_in_month':value

			},
			callback: function(res) {
				if (!res.exc) {
					// code snippet
					console.log(res.message)
					if (res.message["flag"] == false){
                        frappe.web_form.fields_dict['session_planned_in_month'].df.description = '<p style="color: Tomato;">No of session planned in a month should not be more than ' + res.message["session_planned_in_month"] + ' </p>'
                        frappe.web_form.get_field('session_planned_in_month').refresh()
                    }
                    else if (res.message["flag"] == true){
                        frappe.web_form.fields_dict['session_planned_in_month'].df.description = ''
                        frappe.web_form.get_field('session_planned_in_month').refresh()
                    }
				}
			}
		});
	});
})

File: appln_teacherbot_assam/mod_teacherbot_assam/web_form/session_tracking_tool/session_tracking_tool.py
import frappe

def get_context(context):
	# do your magic here
	pass

@frappe.whitelist(allow_guest=True)
def get_block(district_name):
    dd = frappe.db.sql(
        f"""select name from `tabMBTC_HR_Master_Block` where district = '{district_name}'""", as_dict=1)
    return '\n' + '\n'.join(data_item['name'] for data_item in dd)

@frappe.whitelist(allow_guest=True)
def get_school(district_name, block_name):
    dd = frappe.db.sql(
        f"""select name from `tabMBTC_HR_Master_School` where district = '{district_name}' and block = '{block_name}'""", as_dict=1)
    return '\n' + '\n'.join(data_item['name'] for data_item in dd)

@frappe.whitelist(allow_guest=True)
def validate_planned_session(session_in_month):
    session_planned_in_month = frappe.db.get_single_value("TeacherBot Assam Settings", "session_planned_in_month")
    if int(session_in_month) >  int(session_planned_in_month):
         return {"flag": False, "session_planned_in_month": session_planned_in_month}
    return {"flag": True, "session_planned_in_month": session_planned_in_month}
